openlibrary/plugins/openlibrary/js/page-barcodescanner/LazyBookCard.js

```diff
@@ -105,16 +103,14 @@ export default class LazyBookCard {
                     }
                 }
 
-                return fetch(`${OL_BASE}${editionRecord.works[0].key}.json`).then(r => r.json())
-            }).then(workRecord => {
-                return Promise.all(
-                    workRecord.authors
-                        .map(a => fetch(`${OL_BASE}${a.author.key}.json`).then(r => r.json()))
-                );
-            }).then(authorRecords => {
+                // Split up key, for example /book/00000 becomes 00000.
+                const edition_key = editionRecord.key.split('/').pop();
+
+                return fetch(`${OL_BASE}search.json?q=edition_key:${edition_key}&fields=key,author_name`).then(r => r.json())
+            }).then(searchRecords => {
```

Comment:
Nice work @leanderoudakkertue this looks great! I just realized there's an edge case I missed when I created this issue; sometimes `/isbn` creates new books, and it takes ~a minute for them to appear in search. We don't leave enough time in that case. Can you add a fall back in here? Something like:

```
if (!searchRecords.docs.length) {
    // Do the previous more expensive method of fetching all the author records
}
```

That should let us handle both cases.

Reply:
@cdrini I implemented and tested the fallback.